### PR TITLE
Rename and deprecate XMPPConnection methods

### PR DESCRIPTION
**Pull request by fuentesj11**
Rename and deprecate XMPPConnections methods as described in SMACK-802

### CODE REVIEW

**Comment by fuentesj11** [comment]
I thought the methods were supposed to be deprecated and thus the Werror flag is failing the build.  Am I wrong or do I need to submit this a different way?

File: smack-core/src/main/java/org/jivesoftware/smack/XMPPConnection.java [hunk 1]
```diff
@@ -391,15 +413,38 @@ public StanzaCollector createStanzaCollectorAndSend(StanzaFilter packetFilter, S
...
+    @Deprecated
     public void addPacketInterceptor(StanzaListener packetInterceptor, StanzaFilter packetFilter);
 
+    /**
+     * Registers a stanza(/packet) interceptor with this connection. The interceptor will be
+     * invoked every time a stanza(/packet) is about to be sent by this connection. Interceptors
+     * may modify the stanza(/packet) to be sent. A stanza(/packet) filter determines which packets
+     * will be delivered to the interceptor.
+     *
+     * <p>
```
Comment (vanitasvitae) [inline smack-core/src/main/java/org/jivesoftware/smack/XMPPConnection.java]: I think you missed a </p> here.

Comment (fuentesj11) [reply smack-core/src/main/java/org/jivesoftware/smack/XMPPConnection.java]: Thank you I completely missed that!

**Comment by vanitasvitae** [comment]
>I thought the methods were supposed to be deprecated and thus the Werror flag is failing the build. Am I wrong or do I need to submit this a different way?

The build is probably failing, because various pieces of code still point to the now deprecated method. The proper way around this would *probably* be to change all calls to the old method to instead call the renamed method.

@Flowdalic knows best what to do though :)

File: smack-core/src/test/java/org/jivesoftware/smack/test/util/CharSequenceEquals.java [hunk 2]
```diff
@@ -1,47 +0,0 @@
-/**
```
Comment (vanitasvitae) [inline smack-core/src/test/java/org/jivesoftware/smack/test/util/CharSequenceEquals.java]: Why did you delete this class?

**Comment by fuentesj11** [comment]
Sorry about all those errors.  I should have been much better at that.

**Comment by vanitasvitae** [comment]
You can check for errors using `gradle check`.

**Comment by fuentesj11** [comment]
Thank you!  I literally had everything that could wrong with gradle in Intellij happen.

File: smack-core/src/test/java/org/jivesoftware/smack/test/util/CharSequenceEquals.java (hunk 2 above)
Comment (Flowdalic) [reply smack-core/src/test/java/org/jivesoftware/smack/test/util/CharSequenceEquals.java]: Yep, why did this class go? If it's not longer used then we should consider removing it, but with an extra commit.

**Comment by fuentesj11** [comment]
Is anyone aware there is a CharsequenceEquals and CharSequenceEquals class?

**Comment by vanitasvitae** [comment]
>Is anyone aware there is a CharsequenceEquals and CharSequenceEquals class?

Yeah, looks like the `CharsequenceEquals` class can be removed.

**Comment by fuentesj11** [comment]
It that an issue/ticket/task in itself?  I just looked through all branches and it appears to be in all branches.

**Comment by Flowdalic** [comment]
Squash please
